buildspec_load_test.yml

```diff
@@ -28,10 +28,16 @@ phases:
   build:
     commands:
       # Assume role to get the credentials, and set them as environment variables
-      - CREDS=`aws sts assume-role --role-arn ${LOAD_TEST_CFN_ROLE_ARN} --role-session-name load-test-cfn`
-      - export AWS_ACCESS_KEY_ID=`echo $CREDS | jq -r .Credentials.AccessKeyId`
-      - export AWS_SECRET_ACCESS_KEY=`echo $CREDS | jq -r .Credentials.SecretAccessKey`
-      - export AWS_SESSION_TOKEN=`echo $CREDS | jq -r .Credentials.SessionToken`
+      # Will automatically refresh credentials after 1 hour for 5 hours if needed.
+      - |
+        for i in {1..5}
+        do wait
+        CREDS=`aws sts assume-role --role-arn ${LOAD_TEST_CFN_ROLE_ARN} --role-session-name load-test-cfn`
+        export AWS_ACCESS_KEY_ID=`echo $CREDS | jq -r .Credentials.AccessKeyId`
+        export AWS_SECRET_ACCESS_KEY=`echo $CREDS | jq -r .Credentials.SecretAccessKey`
+        export AWS_SESSION_TOKEN=`echo $CREDS | jq -r .Credentials.SessionToken`
+        sleep 3590
+        done &
```

Comment:
won't this block the other code from running? (the code below this) until the while loop finishes? Or does the `do wait` bit do something I don't know about where this is a sub-process?

Reply:
this was a subprocess. because I added the & at the end of the command. It doesn't work great because python caches the credentials.

Had to remove this system.